qhub/template/{{ cookiecutter.repo_directory }}/image/jupyterlab/environment.yaml

```diff
@@ -2,39 +2,58 @@ name: base
 channels:
   - conda-forge
 dependencies:
- # general
- - pip
+  # general
+  - pip
+  - setuptools
 
- # jupyterhub/jupyterlab
- - nb_conda_kernels
- - ipython > 7
- - jupyter-server-proxy
- - jupyterlab-git
- - ipywidgets
- - jupyterlab >= 2
- - jupyter_client
- - jupyter_console
- - jupyterhub
- - nodejs >= 10
- - nbconvert
- - nbval
- - dask_labextension
+  # jupyterhub/jupyterlab
+  - nb_conda_kernels
+  - ipython > 7
+  - jupyter-server-proxy
+  - jupyterlab-git
+  - ipywidgets
+  - jupyterlab >= 3
+  - jupyter_client
+  - jupyter_console
+  - jupyterhub
+  - nodejs >= 10
+  - nbconvert
+  - nbval
 
- # viz tools
- - panel
- - param
- - python-graphviz
- - ipyleaflet
+  # dask
+  - dask_labextension # requires dask
+  - dask==2.14.0
+  - distributed==2.14.0
+  - dask-gateway==0.6.1
+  - pandas
 
- # testing, docs, linting
- - pytest
- - hypothesis
- - flake8
- - sphinx
- - pytest-cov
- - black
- - isort
- - importnb
+  # viz tools
```

Comment:
Genuine question: why are we not pinning viz dependencies but Jlab/Dask ones are?

Reply:
Good question.  I did not really put much thought into what was pinned, I pretty much just updated versions for things that were previously pinned.  
I added dask as an explicit dependency to the jupyterlab environment because it was being installed indirectly anyway, and it needed to be pinned for compatibility with the version of dask-gateway qhub is currently using.  The dask pins will need to be updated or preferrably removed once dask-gateway is updated.

Reply:
Gotcha - I totally see the importance of pinning deps for compatibility. 

Though it would be good to also pin the viz and update periodically to breaking changes to creep in silently for our users

Reply:
dask and distributed are things we really need to pin since mismatched in those between jlab node and dask workers has been the cause of many subtle and not so subtle bugs in using dask. We really want to handle pinning through a qhub metapackage though https://github.com/Quansight/qhub-cloud/issues/86